scene/resources/curve.h

```diff
@@ -250,6 +250,9 @@ class Curve2D : public Resource {
 class Curve3D : public Resource {
 	GDCLASS(Curve3D, Resource);
 
+	friend class Path3DGizmo;
+	friend class Path3DEditorPlugin;
+
```

Comment:
Do you want to get something from Curve3D's private methods/props? Probably this should not be necessary. If you can't get the information you need, just expose the getter in Curve3D. If this is really necessary (I don't think so), you should use `ifdef TOOLS_ENABLED`.

Reply:
> Do you want to get something from Curve3D's private methods/props?

Yes

> If you can't get the information you need, just expose the getter in Curve3D.

Ok, I'll go this way.